test/hotspot/jtreg/compiler/vectorization/TestVectorizationNegativeScale.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8332827
+ * @summary [REDO] C2: crash in compiled code because of dependency on removed range check CastIIs
```

Comment:
Can you say what happened with this test before your fix? Did it not vectorize? Or crash?

Reply:
It's a test case for a failure I ran into while running test. The root of the failure is some other bug that got fixed in the meantime. But the failure with this particular test case only happens with the change for this PR (and without the bug fix for the other bug). So while the test is not about this PR per se, I thought was interesting to keep the test anyway.